### DISCUSSION TITLE
Quasar SSR Mode "Document Is Not Defined" Error

### DISCUSSION THREAD

**Discussion by mecjos** [quasarframework/quasar#12876] [CLI - SSR mode] [answered]
Hi.. I was working on project without problem then after a while restarted computer and get document is not defined error..

![image](https://user-images.githubusercontent.com/11603766/159336780-d8f71c41-d5e8-40c2-af89-3705b2687a06.png)

I couldn't understand what happened.. just restarted.. Can anyone help?

**Comment by MilosPaunovic** [comment]
Just for a sanity check do remove `node_modules` and `package-lock.json` and do a fresh dependency install, see if it helps.

**Reply by mecjos** [reply]
didn't work.. same error.. It's already a fresh installation.. I didn't change anything or add any new package.. That's really wierd... only gives error on ssr dev mode.

**Reply by mecjos** [reply]
it gives following lines in console.. and I think there are more than one error... r

![image](https://user-images.githubusercontent.com/11603766/159369224-b256a9cb-136a-43ee-b0ba-9269ff1b7a7e.png)


```
[Vue warn]: A plugin must either be a function or an object with an "install" function.
[Quasar Dev Webserver] / -> error during render
ReferenceError: document is not defined
    at normalizeContainer (C:\Users\yusuf\Desktop\webdev\sortup\landing\node_modules\@vue\runtime-dom\dist\runtime-dom.cjs.js:1605:21)      
    at Object.app.mount (C:\Users\yusuf\Desktop\webdev\sortup\landing\node_modules\@vue\runtime-dom\dist\runtime-dom.cjs.js:1526:27)        
    at Array.<anonymous> (render-app.js:260:7)
    at render-app.js:2325:27
```

**Comment by rstoenescu** [comment, marked as answer]
We can only help if you provide a reproduction repo.
By the looks of the error, you are wrongly installing a Vue plugin.

**Reply by mecjos** [reply]
Yes you are right..  I tried to embed aos animations into quasar. following boot file coused the problem.. still I couldn't find how I can use aos scroll animations in quasar. 

```
import { boot } from 'quasar/wrappers'
import AOS from 'aos'
import 'aos/dist/aos.css'

 "async" is optional;
 more info on params: https://v2.quasar.dev/quasar-cli/boot-files
 export default boot(({ app, ssrContext } ) => {
  app.use(AOS)
   app.mount('#app')
 })
```

**Reply by hawkeye64** [reply]
It's likely the AOS you are using is made for Vue v2, not Vue v3. Check their repo. Check if they have a specific branch for Vue v3.

**Reply by mecjos** [reply]
I followed the instruction below 
https://egghead.io/blog/how-to-use-the-animate-on-scroll-aos-library-in-vue

it's described to be used in vue but i think I need to do something else for quasar.. it seems to me my mounting method is problematic.

**Reply by hawkeye64** [reply]
Also, don't mount the app. This will happen automatically after all the boot files are initialized.
